tools/topology/sof/pipe-smart-amplifier-playback.m4

```diff
@@ -137,5 +137,5 @@ indir(`define', concat(`PIPELINE_PCM_', PIPELINE_ID), Smart Amplifier Playback P
 #
 
 # PCM capabilities supported by FW
-PCM_CAPABILITIES(Smart Amplifier Playback PCM_ID, CAPABILITY_FORMAT_NAME(PIPELINE_FORMAT), 48000, 48000, 2, PIPELINE_CHANNELS, 2, 16, 192, 16384, 65536, 65536)
+PCM_CAPABILITIES(Smart Amplifier Playback PCM_ID, CAPABILITY_FORMAT_NAME(PIPELINE_FORMAT), 48000, 48000, SMART_PB_CH_NUM, SMART_PB_CH_NUM, 2, 16, 192, 16384, 65536, 65536)
 
```

Comment:
it's now becoming rather hard to figure out that the rest of the information means. 2 as in 2 channels or 2 periods. 16 what? 192 what unit. the rest seems to be bytes but it's confusing.

Reply:
@plbossart see the function declaration in the pcm.m4 here:

```
dnl PCM_CAPABILITIES(name, formats, rate_min, rate_max, channels_min, channels_max, periods_min, periods_max, period_size_min, period_size_max, buffer_size_min, buffer_size_max)
```

Reply:
You missed the point @keyonjie. Humans are not wired to look at 13 parameters on the same line. Adding comments and newlines is much easier to visualize.

Reply:
Got your point now @plbossart. That's the fact that when we are playing with m4 language. We can add this "dnl PCM_CAPABILITIES(...)" line to each .m4 file where the PCM_CAPABILITIES() is used (grepped and found there are 55 in total), just like what we have been doing for "PIPELINE_PCM_ADD()". But IMHO that is actually quite risky, imagine the interface is changed, we have to update all 55 declares, otherwise, some parameters might be used wrongly and nobody will find it.

If we treat these m4 files as C coding, keeping a single declare of a helper function makes sense to me, people who is using the helper should check the declare (in pcm.m4 in our example here) to understand each parameter.

Reply:
@keyonjie I think what @plbossart means is something like
```
PCM_CAPABILITIES(Smart Amplifier Playback PCM_ID,
    CAPABILITY_FORMAT_NAME(PIPELINE_FORMAT), 
    48000, 48000, dnl some comment here 
    SMART_PB_CH_NUM, SMART_PB_CH_NUM, dnl some comment here 
    2, 16, 192,  dnl another comment here etc
    16384, 65536, 65536)
```
This ways it's easier to see. Btw please just do this one, no need to update all other files here.